grouped-categories.js

```diff
@@ -583,29 +587,78 @@
 			return ret;
 		}
 
+		// #220
+		let slotWidth = horiz ?
+			(axis.width / axis.categories.length) * tick.parent.leaves :
+			(axis.height / axis.categories.length) * tick.parent.leaves;
+
+		if (axis.options.labels.step === 1 && tick.label) {
+			if (slotWidth < 15) {
+				tick.label.css({ display: 'none' });
+			} else if ((tick.rotation !== -90 || tick.rotation !== 90) &&
+				tick.label.getBBox().width < slotWidth) {
+				tick.label.css({
+					display: 'block',
+					width: slotWidth,
+					whiteSpace: 'normal',
+					textOverflow: 'ellipsis'
+				});
+			} else {
+				tick.label.css({
+					display: 'block',
+					whiteSpace: undefined,
+					textOverflow: undefined
+				});
+			}
+		}
 
 		while (group.parent) {
 			group = group.parent;
-			
+
 			var fix = fixOffset(treeCat),
 				userX = group.labelOffsets.x,
 				userY = group.labelOffsets.y;
-			
+
 			minPos = tickPosition(tick, mathMax(group.startAt - 1, min - 1));
 			maxPos = tickPosition(tick, mathMin(group.startAt + group.leaves - 1 - fix, max));
 			bBox = group.label.getBBox(true);
 			lvlSize = axis.groupSize(depth);
+
+			// #220
+			let groupSlotWidth = horiz ?
+				(axis.width / axis.categories.length) * group.leaves :
+				(axis.height / axis.categories.length) * group.leaves;
+
+			if (axis.options.labels.step === 1 && group.label) {
+				if (groupSlotWidth < 15) {
+					group.label.css({ display: 'none' });
+				} else if ((tick.rotation !== -90 || tick.rotation !== 90) &&
+					tick.label.getBBox().width < slotWidth) {
+					group.label.css({
+						display: 'block',
+						width: groupSlotWidth,
+						textOverflow: 'ellipsis'
+					});
+				} else {
+					group.label.css({
+						display: 'block',
+						width: undefined,
+						textOverflow: undefined
+					});
+				}
+			}
```

Comment:
This logic, and the one above seems to be 99% the same - can we move it to the function?

Reply:
Thanks! Sure, i will do that